microprofile-opentracing/src/test/java/org/jboss/eap/qe/microprofile/opentracing/v10/MicroProfileOpenTracing10Test.java

```diff
@@ -0,0 +1,173 @@
+package org.jboss.eap.qe.microprofile.opentracing.v10;
+
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.junit.InSequence;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.eap.qe.microprofile.opentracing.MicroProfileOpenTracingServerHelper;
+import org.jboss.eap.qe.ts.common.docker.Docker;
+import org.jboss.eap.qe.ts.common.docker.DockerContainers;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import io.opentracing.Tracer;
+import io.restassured.RestAssured;
+
+@RunWith(Arquillian.class)
+public class MicroProfileOpenTracing10Test {
+
+    private static final String APPLICATION_NAME = MicroProfileOpenTracing10Test.class.getSimpleName();
+
+    @ClassRule
+    public static Docker jaegerContainer = DockerContainers.jaeger();
+
+    @ArquillianResource
+    private URL baseApplicationUrl;
+
+    @BeforeClass
+    public static void reloadServer() throws Exception {
+        MicroProfileOpenTracingServerHelper.reload();
+    }
+
+    @Deployment
+    public static Archive<?> deployment() {
+        String mpConfig = "sampler.param=1\n" + // sample every trace
+                "sampler.type=const\n"; // use the same decision strategy for all traces
+
+        return ShrinkWrap.create(WebArchive.class, APPLICATION_NAME + ".war")
+                .addPackage(RestApplication.class.getPackage())
+                .addPackage(Tracer.class.getPackage())
+                .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml")
+                .addAsManifestResource(new StringAsset(mpConfig), "microprofile-config.properties");
+    }
+
+    /**
+     * @tpTestDetails Send a request to deployed application on /rest endpoint.
+     * @tpPassCrit Deployed application is available and returns expected response.
```

Comment:
Could you provide more information in each JavaDoc, as it's really hard to say what is tested from description?

Reply:
updated, please let me know whether something is still not understandable, thanks